Add kangax runner

Achie72 opened the pull request:
The runner clones down the files into test/kangax,
patches it with escargot, copies the runner into it,
and finally it runs and prints the results.

kangaxES6PLUSRes,kangaxES6Res and kangaxRes files are created to store the corresponding suite's full log. The summary only contains the original kangax summary, and all the failing test.

`--suite {es6,es6plus}` addded as parameters, default run without any parameters will run both suites.

Signed-off-by: Bela Toth tbela@inf.u-szeged.hu

Achie72 commented:
So, i have some questions/feedback requests about this pull request.

- The escargot.js is just a modified JerryScript runner from the orignial compat-table repository, as such i don't know how to handle copyright issues there. Should i mention a co-author?
- The [enviroments](https://github.com/Samsung/escargot/pull/599/files#diff-b351748fa39ad006cb22eacacc36ad7dR9) patch is lacking some information from official owners, so i wanted to ask about this. The following information are "needed": a version number for the first line `"escargot:" {` as the compat table uses previous version results too, but atm, i'm bypassing it (hence the `return 1`). That version number is then present also in the `full": "Escargot <version>:` part, atm i used master. `"family"`,` "short"`, and `"release"` are also information beyond what i should define. All these are not needed as long as we want to use this in our repository, but for the official one, when we want to push data into the kangax compat-table repository, we will be needing these.
- Atm there is no cleanup phase, i did not see it as necessary, but i will add it if requested.
- Errors in the `node escargot.js escargot` call (which is running the  suite) are only present in the corresponding log files. I need some feedback on this as well, is it good this way, or i should push it into the python output?
- The fail output is in raw form. For a more refined one, some more scripts would be needed, but i think they are readable as they are now. If requested i will create those as well.

clover2123 commented:
* I have installed `node` in Jenkins server. Please retry CI.
* How about adding kangax repo in a form of git module under `test` directory like test262?

> The escargot.js is just a modified JerryScript runner from the orignial compat-table repository, as such i don't know how to handle copyright issues there. Should i mention a co-author?

IMO if `escargot.js` is first originated from kangax repo, we don't need to mentain co-authors.
But to be sure, let me know the location of origin file in kangax and jerryscript repo both.
I'll check it.

> The enviroments patch is lacking some information from official owners

We can't define information for environments now because we should first fix some bugs about kangax TCs before uploading to the official kangax repo (we may release a new version of Escargot).
So, please fill the environments with temporal values (it seems fine now).

> Errors in the node escargot.js escargot call (which is running the suite) are only present in the corresponding log files. I need some feedback on this as well, is it good this way, or i should push it into the python output?

It would be better to print output as well as log files. Because we can easily check the result on CI.

Achie72 commented:
> How about adding kangax repo in a form of git module under test directory like test262?  

I added it as a submodule under the name of kangax, as the compat-table name is a bit hard to grasp from first glance. The diff shows it to me as it has a different indentation, but in the text editor i see no difference.

> IMO if escargot.js is first originated from kangax repo, we don't need to mention co-authors.
But to be sure, let me know the location of origin file in kangax and jerryscript repo both.
I'll check it.

@clover2123 the original file is located here: [Link](https://github.com/kangax/compat-table/blob/gh-pages/jerryscript.js), i modified this, which seems to work perfectly.

> It would be better to print output as well as log files. Because we can easily check the result on CI.

Now the script creates new files for the errors, and also prints them in the log.

clover2123 commented:
@Achie72 
Could you check the error on CI result?
It seems that there is a problem in loading `escargot.js` file.

Achie72 commented:
So it seems that it works, i finally managed to fix the problems. Tell me if you need anything else from this PR.

clover2123 commented on tools/kangax/run-kangax.py:
```diff
@@ -0,0 +1,142 @@
+    else:
+        res = open("./kangaxRes.txt", "w")
+        nodeFails = open("./kangaxRunErrors.txt", "w")
+        subprocess.call(['node', './escargot.js', engine, suite], stdout = res)
```
It seems that `nodeFailes` is omitted for stderr here.

Achie72 replied:
That's a good catch, i missed this one!

clover2123 commented:
Thank you for your great patch!
Are you going to implement `_onExit` feature as well?

Achie72 commented:
Yes, i'm working on it atm, but it's tricky a bit.
I may not have been clear with that "feature" as well, as it will only go into the Shell and nowhere else.

clover2123 commented:
@Achie72 
it would be better to add the new feature(_onExit) under `ESCARGOT_ENABLE_TEST` macro in `Shell.cpp`.
https://github.com/Samsung/escargot/blob/295c4c21e5e3be15770217f1ae7a0a2d5bc7e9b2/src/shell/Shell.cpp#L358
